test/functional/p2p_addr_relay.py

```diff
@@ -32,11 +37,15 @@
 
 
 class AddrReceiver(P2PInterface):
+    num_ipv4_received = 0
+
     def on_addr(self, message):
         for addr in message.addrs:
             assert_equal(addr.nServices, 9)
+            assert_greater_than(8343, addr.port)
+            assert_greater_than_or_equal(addr.port, 8333)
```

Comment:
<details><summary>a few suggestions for the test code</summary><p>

```diff
@@ -18,8 +18,6 @@ from test_framework.mininode import (
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import (
     assert_equal,
-    assert_greater_than,
-    assert_greater_than_or_equal,
 )
 import time
 
@@ -42,8 +40,8 @@ class AddrReceiver(P2PInterface):
     def on_addr(self, message):
         for addr in message.addrs:
             assert_equal(addr.nServices, 9)
-            assert_greater_than(8343, addr.port)
-            assert_greater_than_or_equal(addr.port, 8333)
+            if not 8333 <= addr.port < 8343:
+                raise AssertionError("Invalid addr.port of {} (8333-8342 expected)".format(addr.port))
             assert addr.ip.startswith('123.123.123.')
             self.num_ipv4_received += 1
 
@@ -59,14 +57,14 @@ class AddrTest(BitcoinTestFramework):
         msg = msg_addr()
 
         self.log.info('Send too-large addr message')
-        msg.addrs = ADDRS * 101 # more than 1000 addresses in one message
+        msg.addrs = ADDRS * 101  # more than 1000 addresses in one message
         with self.nodes[0].assert_debug_log(['addr message size = 1010']):
             addr_source.send_and_ping(msg)
 
         self.log.info('Check that addr message content is relayed and added to addrman')
         num_receivers = 7
         receivers = []
-        for i in range(num_receivers):
+        for _ in range(num_receivers):
             receivers.append(self.nodes[0].add_p2p_connection(AddrReceiver()))
         msg.addrs = ADDRS
-        with self.nodes[0].assert_debug_log([
-                'Added {} addresses from 127.0.0.1: 0 tried'.format(num_ipv4_addrs),
-                'received: addr (301 bytes) peer=0',
-        ]):
+        with self.nodes[0].assert_debug_log(
+            [
+                "Added {} addresses from 127.0.0.1: 0 tried".format(num_ipv4_addrs),
+                "received: addr (301 bytes) peer=0",
+            ]
+        ):
             addr_source.send_and_ping(msg)
```
</p></details>